### DISCUSSION TITLE
Remapping of Curvilinear data (COSMO) on regular Sentinel 5P grid

### DISCUSSION THREAD

**Discussion by sauerland154** [pydata/xarray#7396] [General]
Hello, 
I am a PhD student at the German Aerospace Center in Munich. I have an xarray DataArray with COSMO model data. My goal is to remap these data with 2d coordinates geolon (rlon,rlat) and geolat (rlon,rlat) onto the regular Satellite grid (Sentinel5P) where longitude(scaline, ground_pixel) and latitude(scaline,ground_pixel) are the Sentinel5P coordinates.

In order to do so, I don't want to apply an interpolation or a built-in remapping with xesmf. With this the model values will change. I want a remapping method, where the mass of the COSMO model data in my xarray DataArray will be conserved.

I've wrote a little function, which should show exactly what I want to do. But this takes to long in a for loop. I really want to apply it on the whole xarray DataArray at once so vectorised:

```python
def naive_fast(geolon, geolat, lat0, lon0): #lat0, lon0 #(Satellite Target Coordinates of Sentinel5P)
 
    # Read latitude and longitude from file into numpy arrays                                                                                                    
    latvals = geolat[:] #latitude of COSMO data
    lonvals = geolon[:] #longitude of COSMO data 
    dist_sq = (latvals-lat0)**2 + (lonvals-lon0)**2
    minindex_flattened = dist_sq.argmin()  # 1D index of min element                                                                                             
 
    iy_min,ix_min = np.unravel_index(minindex_flattened, latvals[0,:,:].shape)
 
    return iy_min,ix_min
```
for loop:
```python
(c_y, c_x) = naive_fast(lat.values, lon.values, combined.lat[i].values, combined.lon[i].values)
    #print(i)                                                                                                                                                                                          
final_NO2[:,i]=*tracer_gp_NO2[:,c_y,c_x].values*1E9
```

The ultimate goal is to find the position for each tracer_gp_NO2 (curvilinear) on the regular satellite grid without changing the order of magnitude of the tracer_gp_NO2 mixing ratio.

So do you maybe know, if there is an out of the box solution for xarray for such a problem? If you have any questions do not hesitate to comment this thread. 
Cheers, Markus.

**Comment by keewis** [comment]
isn't this just a nearest neighbor interpolation? If so, you should be able to use `xesmf`'s `"nearest_s2d"` or `"nearest_d2s"` method, and multiply the result by `1e9`.
